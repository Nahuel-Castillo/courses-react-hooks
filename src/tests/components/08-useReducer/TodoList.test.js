import '@testing-library/jest-dom';
import TodoList from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";
import { shallow } from 'enzyme';

describe('Test of <TodoList/>', () => {
   
    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(
        <TodoList
            todos={ demoTodos }
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />
    );

    test('should show correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('should have 2 <TodoListItem/>', () => {
        
        expect( wrapper.find('TodoListItem').length ).toBe( demoTodos.length );

        expect( wrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual( expect.any( Function) );
    });
});