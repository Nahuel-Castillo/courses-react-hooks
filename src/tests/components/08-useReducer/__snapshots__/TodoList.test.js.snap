// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <TodoList/> should show correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Learn React",
        "done": true,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Learn Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
